x/upgrade/alias.go

```diff
@@ -33,6 +33,8 @@ var (
 	NewQueryAppliedParams            = types.NewQueryAppliedParams
 	NewKeeper                        = keeper.NewKeeper
 	NewQuerier                       = keeper.NewQuerier
+	ConvertIntArrayToInt64           = types.ConvertIntArrayToInt64
```

Comment:
I don't think we need any of these aliases or utility functions.

Reply:
> Looking good @anilCSE -- left a few more comments 
> 
> Question: `skipUpgradeHeights` are provided to the module from user CLI input, no? This array/map dictates the control flow of `BeginBlock` (in that it clears the plan). What if different nodes set (or dont set at all) different values of `skipUpgradeHeights`? Is this OK?



> Looking good @anilCSE -- left a few more comments 
> 
> Question: `skipUpgradeHeights` are provided to the module from user CLI input, no? This array/map dictates the control flow of `BeginBlock` (in that it clears the plan). What if different nodes set (or dont set at all) different values of `skipUpgradeHeights`? Is this OK?

Case - 1:
We're checking this only if there's an upgrade plan to be executed at that height, so it doesn't care if a set of heights which don't have upgrade scheduled are passed. So if user passes bunch of random heights,there will be no effect.
Case - 2:
This implemented solution can be used only to skip upgrades which have passed a proposal. So it's in validators intrest that they chose to upgrade at that height or not based on how social consensus on a particular upgrade is. If they chose otherwise, they'll not be able to sync or sign the blocks after that height. Either the ways, it shouldn't be a problem